### PR TITLE
Update Wasmtime's WASI documentation.

### LINKED ISSUE #9107: Dead link in WASI tutorial
Hi! The WASI tutorial contains a dead link to the browser polyfill:

https://github.com/bytecodealliance/wasmtime/blob/22a4480e90d2e5bb4933a0bd027bd31d007b8b04/docs/WASI-tutorial.md?plain=1#L323

### PR DESCRIPTION
**Pull request by sunfishcode**
Replace most of Wasmtime's original WASI documentation with links to other resources that are now available.

This documentation was written when WASI was first released, and while many of the big ideas are still present in WASI, many of the details have changed. For example, [Jco] is now a much better way to run WASI programs in Browsers than the original Web polyfill, and WASI is now goverened by the WASI Subgroup rather than by Wasmtime itself, and so on.

[Jco]: https://github.com/bytecodealliance/jco

Fixes #9107.

### CODE REVIEW

File: docs/WASI-overview.md [hunk 1]
```diff
@@ -1,163 +1,9 @@
...
-could take these ideas even further allowing easier and tighter
-integration between .wasm modules importing WASI and the Web.
-
-## Work in Progress
-
-WASI is currently experimental. Feedback is welcome!
-
-[Wasmtime]: https://github.com/bytecodealliance/wasmtime
+[WASI Subgroup]: https://github.com/WebAssembly/wasi
+[wasi.dev]: https://wasi.dev/interfaces
```
Comment (cfallin) [inline docs/WASI-overview.md:9]: I think this link name should be `[Interfaces page on wasi.dev]` to match the reference above?

**Review by cfallin — APPROVED** [review]
:+1:

File: docs/WASI-overview.md (hunk 1 above)
Comment (sunfishcode) [reply docs/WASI-overview.md]: Yep, fixed.

File: docs/WASI-documents.md [hunk 2]
```diff
@@ -1,28 +1,5 @@
...
-For background on some of the design decisions in WASI, see [the rationale](WASI-rationale.md).
-
-For documentation of the exports required of programs using, see
-[the application ABI](https://github.com/WebAssembly/WASI/blob/main/legacy/application-abi.md).
-
-For some ideas of things that we may want to change about WASI in the
-short term, see the [possible changes](WASI-some-possible-changes.md) document.
-For longer-term ideas, see the [possible future features](WASI-possible-future-features.md)
-document.
+[wasi.dev]: https://wasi.dev/resources
```
Comment (iawia002) [inline docs/WASI-documents.md:5]:
```suggestion
[Resources page on wasi.dev]: https://wasi.dev/resources
```

One more link name typo.

Comment (sunfishcode) [reply docs/WASI-documents.md]: Thanks, fixed.

**Pull request merged**